### DISCUSSION TITLE
MongoDB, multiple databases and relationships

### DISCUSSION THREAD

**Discussion by edorgeville** [mikro-orm/mikro-orm#1985] [Q&A] [answered]
Hi,
I am in the process of refactoring a NodeJS + MongoDB backend. I am trying to fit the existing data into MikroORM. 
An important detail about this existing app is that there is multiple seperate databases referencing one another.

Here is a simplified schema:
- `main`: `database`
  - `user`: collection
    - `_id`: `ObjectId`
    - `name`: `string`
- `productA`: `database`
  - `product`: `collection`
    - `_id`: `ObjectId`
    - `user_id`: `Object` (references `main.users._id`)
- `productB`: `database`
  - `product`: `collection`
    - `_id`: `ObjectId`
    - `user_id`: `ObjectId` (references `main.users._id`)
- `productC`: `database`
- etc.

I figured out how to connect to different databases using different MikroORM instances.
If I try to [make a relationship from `productA` to `user`](https://github.com/edorgeville/mikro-orm-mongodb-multi-databases/blob/ea3de945ed759ca34cea4d90ea9a33d036d86a4c/src/entities/productA/Product.ts#L14-L15), I get the following error:
```
mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\errors.js:142
  return new MetadataError(`Entity '${className}' was not discovered, please make sure to provide it in 'entities' array when initializing the ORM (used in ${source})`);
         ^
MetadataError: Entity 'User' was not discovered, please make sure to provide it in 'entities' array when initializing the ORM (used in Product.user)
at Function.fromUnknownEntity (mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\errors.js:142:16)
at mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\metadata\MetadataValidator.js:49:46
at Array.forEach (<anonymous>)
at mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\metadata\MetadataValidator.js:47:67
at Array.forEach (<anonymous>)
at MetadataValidator.validateDiscovered (mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\metadata\MetadataValidator.js:47:20)
at MetadataDiscovery.findEntities (mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\metadata\MetadataDiscovery.js:81:24)
at processTicksAndRejections (node:internal/process/task_queues:94:5)
at MetadataDiscovery.discover (mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\metadata\MetadataDiscovery.js:34:9)
at Function.init (mikro-orm-mongodb-multi-databases\node_modules\@mikro-orm\core\MikroORM.js:42:24)
```
I made an example repo to reproduce my issue: https://github.com/edorgeville/mikro-orm-mongodb-multi-databases

So my question is: is it possible to make a reference between two seperate MongoDB databases?
Thanks!

**Comment by B4nan** [comment, marked as answer]
Nope, not supported in mongo. Related issue from today: #1971 

And even with that, you definitely can't make a relation to an entity that is not part of the ORM configuration. Correct way would be to use single instance and `FindOptions.schema` (which is not supported in mongo just yet).

**Reply by B4nan** [reply]
But if you are fine with multiple ORM instances, it might be just fine if you let all of them discover all the entities.

**Reply by edorgeville** [reply]
Discovering all entities... works! Perfect.
Thanks a lot for your blazing fast answer 🚀

Updated my repo for posterity https://github.com/edorgeville/mikro-orm-mongodb-multi-databases/commit/a42c377eaef58631cc8f26f99965ae7180b4e175
